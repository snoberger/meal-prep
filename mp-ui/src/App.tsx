import React from "react";
import "./App.css";
import Routes from "./Routes";
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import rootReducer from './store/rootReducer';

const store = createStore(rootReducer, undefined, undefined);

class App extends React.Component {
  render() {
    return (
      <Provider store={store}>
        <div className="App">
          <Routes></Routes>
        </div>
      </Provider>
    );
  }
}

export default App;
