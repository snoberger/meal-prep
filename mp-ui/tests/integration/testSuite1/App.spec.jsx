import React from 'react';
import App from '../../../src/App';
import { mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
// import {createSerializer} from 'enzyme-to-json';
 

//Used for snapshot
// expect.addSnapshotSerializer(createSerializer({mode: 'deep'}));


describe("App renders", () => {
    it("should match snapshot", () => {
        // TODO have the user be logged in before mount
        const wrapper = mount(
            <MemoryRouter initialEntries={[ '/' ]}>
                <App></App>
            </MemoryRouter>
          );
          expect(wrapper).toBeDefined();
        //expect(wrapper).toMatchSnapshot();
    });
});

