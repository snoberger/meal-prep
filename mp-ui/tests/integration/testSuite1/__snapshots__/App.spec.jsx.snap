// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders should match snapshot 1`] = `
<div
  className="App"
>
  <div>
    <div
      className="login-div"
    >
      <h3
        className="MuiTypography-root MuiTypography-h3"
      >
        Login
      </h3>
      <h6
        className="MuiTypography-root MuiTypography-subtitle1"
      >
        Welcome back! Login to begin meal prepping!
      </h6>
      <div
        className="MuiPaper-root MuiCard-root login-card MuiPaper-elevation1 MuiPaper-rounded"
      >
        <form>
          <div
            className="MuiCardContent-root"
          >
            <div
              className="MuiFormControl-root MuiTextField-root login-input"
            >
              <label
                className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled"
                data-shrink={false}
                htmlFor="email"
                id="email-label"
              >
                Email Address
              </label>
              <div
                className="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-formControl"
                onClick={[Function]}
              >
                <input
                  aria-invalid={false}
                  autoFocus={false}
                  className="MuiInputBase-input MuiFilledInput-input"
                  disabled={false}
                  id="email"
                  onAnimationStart={[Function]}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              className="MuiFormControl-root MuiTextField-root login-input"
            >
              <label
                className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled"
                data-shrink={false}
                htmlFor="password"
                id="password-label"
              >
                Password
              </label>
              <div
                className="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-formControl"
                onClick={[Function]}
              >
                <input
                  aria-invalid={false}
                  autoFocus={false}
                  className="MuiInputBase-input MuiFilledInput-input"
                  disabled={false}
                  id="password"
                  onAnimationStart={[Function]}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onFocus={[Function]}
                  required={false}
                  type="password"
                  value=""
                />
              </div>
            </div>
            <a
              className="MuiTypography-root MuiLink-root MuiLink-underlineHover forgot-password MuiTypography-colorPrimary"
              href="/forgotpassword"
              onBlur={[Function]}
              onFocus={[Function]}
            >
              Forgot Password?
            </a>
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-contained login-button MuiButton-containedPrimary"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <span
                className="MuiButton-label"
              >
                Login
              </span>
              <span
                className="MuiTouchRipple-root"
              />
            </button>
            <div
              className="signup"
            >
              <a
                aria-current={null}
                aria-disabled={false}
                className="MuiButtonBase-root MuiButton-root MuiButton-contained signup-button MuiButton-containedSecondary"
                href="/signup"
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role="button"
                tabIndex={0}
              >
                <span
                  className="MuiButton-label"
                >
                  Sign up here
                </span>
                <span
                  className="MuiTouchRipple-root"
                />
              </a>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
